linc_convert/modalities/lsm/spool.py

```diff
@@ -0,0 +1,283 @@
+from linc_convert.utils.orientation import center_affine, orientation_to_affine
+from linc_convert.utils.spool import SpoolSetInterpreter
+from linc_convert.utils.zarr.compressor import make_compressor
+from linc_convert.utils.zarr.generate_pyramid import generate_pyramid
+from linc_convert.utils.zarr.zarr_config import ZarrConfig
+
+spool = cyclopts.App(name="spool", help_format="markdown")
+lsm.command(spool)
+
+"""
+Convert a collection of spool .dat files generated by the LSM pipeline into either a NIfTI-Zarr or OME-Zarr.
+
+Example input files can be found at
+https://lincbrain.org/dandiset/000010/draft/files?location=sourcedata%2Frawdata%2Fmicr%2Fsample18_run10__y10_z01_HR&page=1
+"""
+
+
+@spool.default
+def convert(
+        inp: str,
+        *,
+        out: str,
+        zarr_config: ZarrConfig = None,
+        overlap: int = 30,
+        orientation: str = "coronal",
+        center: bool = True,
+        voxel_size: list[float] = (1, 1, 1),
+        **kwargs
+) -> None:
+    """
+    Convert a collection of spool files generated by the LSM pipeline into ZARR.
+
+    Orientation
+    -----------
+    The anatomical orientation of the slice is given in terms of RAS axes.
+
+    It is a combination of two letters from the set
+    `{"L", "R", "A", "P", "I", "S"}`, where
+
+    * the first letter corresponds to the horizontal dimension and
+        indicates the anatomical meaning of the _right_ of the image,
+    * the second letter corresponds to the vertical dimension and
+        indicates the anatomical meaning of the _bottom_ of the image.
+
+    We also provide the aliases
+
+    * `"coronal"` == `"LI"`
+    * `"axial"` == `"LP"`
+    * `"sagittal"` == `"PI"`
+
+    The orientation flag is only useful when converting to nifti-zarr.
+
+    Parameters
+    ----------
+    inp
+        Path to the root directory, which contains a collection of
+        subfolders named `*_y{:02d}_z{:02d}*`, each containing a
+        collection of files named `*spool.dat`.
+    out
+        Path to the output Zarr directory [<INP>.{ome|nii}.zarr]
+    overlap
+        Number of pixels between slices that are overlapped
+    orientation
+        Orientation of the slice
+    center
+        Set RAS[0, 0, 0] at FOV center
+    voxel_size
+        Voxel size along the X, Y and Z dimension, in micron.
+    """
+    zarr_config = utils.zarr.zarr_config.update(zarr_config, **kwargs)
+    chunk: int = zarr_config.chunk[0]
+    compressor: str = zarr_config.compressor
+    compressor_opt: str = zarr_config.compressor_opt
+    nii: bool = zarr_config.nii
+
+    if isinstance(compressor_opt, str):
+        compressor_opt = ast.literal_eval(compressor_opt)
+
+    CHUNK_PATTERN = re.compile(
+        r"^(?P<prefix>\w*)"
+        r"_run(?P<run>[0-9]+)"
+        r"_?"
+        r"_y(?P<y>[0-9]+)"
+        r"_z(?P<z>[0-9]+)"
+        r"(?P<suffix>\w*)$"
+    )
+
+    all_tiles_folders_names = sorted(glob(os.path.join(inp, "*_y*_z*_HR/")))
```

Comment:
Just noting that the way we currently have the data organized on lincbrain.org won't work if we have more that 1 dataset in the `micr` folder.   So we will need to move these `*_y*_z*_HR/` directories within their respective `sub-*` and `ses-*` directories.